plugins/module_utils/dnac.py

```diff
@@ -801,39 +801,66 @@ def assign_device_to_site(self, device_ids, site_name, site_id):
-        self.log("Assigning devices to site before update: {0}, {1}".
-                 format(site_name, str(assign_network_device_to_site)), "INFO")
-        try:
-            response = self.dnac._exec(
-                family="site_design",
-                function='assign_network_devices_to_a_site',
-                op_modifies=True,
-                params=assign_network_device_to_site
-            )
-            self.log("Response from assigning devices to site: {0}, {1}, {2} .".format(
-                site_name, str(assign_network_device_to_site), str(response["response"])), "INFO")
 
-            self.check_tasks_response_status(response, api_name='assign_device_to_site')
-            if self.result["changed"]:
-                return True
-            else:
-                self.msg = "Failed to receive a valid response from site assignment API: {0}, {1}".format(site_name,
-                                                                                                          str(assign_network_device_to_site))
+        if self.get_ccc_version_as_integer() <= self.get_ccc_version_as_int_from_str("2.3.5.3"):
+            try:
+                response = self.dnac_apply['exec'](
+                    family="sites",
+                    function="assign_devices_to_site",
+                    op_modifies=True,
+                    params={
+                        "site_id": site_id,
+                        "payload": device_ids
+                    },
+                )
+
+                self.check_execution_response_status(response, "assign_devices_to_site")
+                if self.status == "success":
+                    self.result["changed"] = True
+                    self.result['msg'] = "Successfully assigned site {0} to device {1}.".format(site_name, str(device_ids))
+                    self.result['response'] = response.get("executionId")
+                    self.log(self.result['msg'], "INFO")
+                return self
+
+            except Exception as e:
+                self.msg = "Error on assign devices to site: {0}".format(str(e))
                 self.log(self.msg, "ERROR")
                 self.status = "failed"
                 self.module.fail_json(msg=self.msg)
+        else:
+            assign_network_device_to_site = {
+                'deviceIds': device_ids,
+                'siteId': site_id,
+            }
+            self.log("Assigning devices to site before update: {0}, {1}".format(site_name,
+                                                                                str(assign_network_device_to_site)), "INFO")
+            try:
+                response = self.dnac._exec(
+                    family="site_design",
+                    function='assign_network_devices_to_a_site',
+                    op_modifies=True,
+                    params=assign_network_device_to_site
+                )
+                self.log("Response from assigning devices to site: {0}, {1}, {2} .".format(
```

Comment:
Received API response from 'assign_network_device_to_site' while assigning devices to site: {0}, {1}, {2}

Reply:
updated.